pkg/v1/content/content.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package content allows you to create images directly from contents.
+package content
+
+import (
+	"archive/tar"
+	"bytes"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+// Image creates a image with the given contents. These images are reproducible and should be consistent.
+func Image(content ...map[string][]byte) (v1.Image, error) {
```

Comment:
What's the thinking behind making this variadic? It opens up the possibility that 2+ maps could define the same key and we'd have to document which wins in that case.

Reply:
> we'd have to document which wins in that case

IIUC each map is a layer, so we'd just need to document the order of these layers.

Reply:
I expect 99% of users will just use a single layer, but it's easier to get it included in the variadic form since if someone wants two layers they can do it. (it also doesn't make the code or function signature that much harder).

Reply:
Also updated the docs.